src/uct/ib/rc/accel/rc_mlx5_ep.c

```diff
@@ -742,13 +747,16 @@ uct_rc_mlx5_ep_connect_qp(uct_rc_mlx5_iface_common_t *iface,
     }
 }
 
-ucs_status_t uct_rc_mlx5_ep_connect_to_ep(uct_ep_h tl_ep,
-                                          const uct_device_addr_t *dev_addr,
-                                          const uct_ep_addr_t *ep_addr)
+ucs_status_t
+uct_rc_mlx5_ep_connect_to_ep_v2(uct_ep_h tl_ep,
+                                const uct_device_addr_t *device_addr,
+                                const uct_ep_addr_t *ep_addr,
+                                const uct_ep_connect_to_ep_params_t *params)
 {
     UCT_RC_MLX5_EP_DECL(tl_ep, iface, ep);
-    const uct_ib_address_t *ib_addr = (const uct_ib_address_t *)dev_addr;
-    const uct_rc_mlx5_ep_address_t *rc_addr = (const uct_rc_mlx5_ep_address_t*)ep_addr;
+    const uct_ib_address_t *ib_addr               = (const uct_ib_address_t *)device_addr;
+    const uct_rc_mlx5_ep_flush_address_t *rc_addr = (const uct_rc_mlx5_ep_flush_address_t*)ep_addr;
```

Comment:
same here - cast to uct_rc_mlx5_ep_flush_address_t* only after checking address length

Reply:
added variable